app/validators/codefendants_validator.rb

```diff
@@ -0,0 +1,42 @@
+class CodefendantsValidator < ActiveModel::Validator
+  attr_reader :record
+
+  def validate(record)
+    @record = record
+
+    record.codefendants.each.with_index do |codefendant, index|
+      add_indexed_errors(codefendant, index) unless codefendant.valid?
+    end
+  end
+
+  private
+
+  def add_indexed_errors(codefendant, index)
+    codefendant.errors.each do |error|
+      attr_name = indexed_attribute(index, error.attribute)
+
+      record.errors.add(
+        attr_name,
+        error.type,
+        message: error_message(codefendant, error), index: index + 1
+      )
+
+      # We define the attribute getter as it doesn't really exist
+      record.define_singleton_method(attr_name) do
+        codefendant.public_send(error.attribute)
+      end
+    end
+  end
+
+  def indexed_attribute(index, attr)
+    "codefendants-attributes[#{index}].#{attr}"
+  end
+
+  # `activemodel.errors.models.steps/case/codefendant_fieldset_form.summary.x.y`
```

Comment:
Is this meant to be here still?

Reply:
You mean the comment? Hmm, it is intended to facilitate understanding what the final computed i18n key is (the one passed to `I18n.t`), because is not so obvious. It helped me figuring out where in the locales I had to place the messages.

Reply:
Ok, all good